### PR TITLE
Add error codes to error messages of Schema loader

### PR DESCRIPTION
**Pull request by brfrn169**
## Description

This PR adds error codes to the error messages of Schema loader.

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/pull/1493

## Changes made

- Added error codes to the error messages of Schema loader.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

Added error codes to the error messages of Schema Loader.

### CODE REVIEW

File: schema-loader/src/main/java/com/scalar/db/schemaloader/command/StorageSpecificCommand.java [hunk 1]
```diff
@@ -105,7 +106,12 @@ private void repairTables(
 
   @VisibleForTesting
   SchemaParser getSchemaParser(Map<String, String> options) throws SchemaLoaderException {
-    return new SchemaParser(schemaFile, options);
+    try {
+      return new SchemaParser(schemaFile, options);
+    } catch (IOException e) {
+      throw new SchemaLoaderException(
+          "Reading the schema file failed; File: " + schemaFile.toAbsolutePath(), e);
```
Comment (komamitsu) [inline schema-loader/src/main/java/com/scalar/db/schemaloader/command/StorageSpecificCommand.java:113]:
```suggestion
          "Reading the schema file failed. File: " + schemaFile.toAbsolutePath(), e);
```
A bit inconsistent with other error messages in `CoreError`?

**Review by komamitsu — APPROVED** [review]
LGTM! Thank you!

File: schema-loader/src/main/java/com/scalar/db/schemaloader/command/StorageSpecificCommand.java (hunk 1 above)
Comment (brfrn169) [reply schema-loader/src/main/java/com/scalar/db/schemaloader/command/StorageSpecificCommand.java:113]: Ah, good catch! Thanks.

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 2]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
       ""),
   DYNAMO_BATCH_SIZE_EXCEEDED(
       Category.USER_ERROR, "0108", "DynamoDB cannot batch more than 100 mutations at once", "", ""),
+  SCHEMA_LOADER_MODIFYING_PARTITION_KEYS_NOT_SUPPORTED(
+      Category.USER_ERROR,
+      "0109",
+      "The partition keys of the table %s.%s were modified, altering them is not supported",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:498]:
```suggestion
      "The partition keys of the table %s.%s were modified, but altering partition keys is not supported",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 3]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      "",
+      ""),
+  SCHEMA_LOADER_MODIFYING_CLUSTERING_KEYS_NOT_SUPPORTED(
+      Category.USER_ERROR,
+      "0110",
+      "The clustering keys of the table %s.%s were modified, altering them is not supported",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:504]:
```suggestion
      "The clustering keys of the table %s.%s were modified, but altering clustering keys is not supported",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 4]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      "",
+      ""),
+  SCHEMA_LOADER_MODIFYING_CLUSTERING_ORDER_NOT_SUPPORTED(
+      Category.USER_ERROR,
+      "0111",
+      "The clustering ordering of the table %s.%s were modified, altering them is not supported",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:510]:
```suggestion
      "The clustering ordering of the table %s.%s were modified, but altering clustering ordering is not supported",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 5]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+  SCHEMA_LOADER_DELETING_COLUMN_NOT_SUPPORTED(
+      Category.USER_ERROR,
+      "0112",
+      "The column %s of the table %s.%s has been deleted. Column deletion is not supported when altering a table",
+      "",
+      ""),
+  SCHEMA_LOADER_MODIFYING_COLUMN_DATA_TYPE_NOT_SUPPORTED(
+      Category.USER_ERROR,
+      "0113",
+      "The data type of the column %s of the table %s.%s was modified, altering it is not supported",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:522]:
```suggestion
      "The data type of the column %s of the table %s.%s was modified, but altering data types is not supported",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 6]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      Category.USER_ERROR,
+      "0116",
+      "Specifying the '--schema-file' option is required when using the '--import' option",
+      "",
+      ""),
+  SCHEMA_LOADER_SPECIFYING_COORDINATOR_WITH_IMPORT_NOT_ALLOWED(
+      Category.USER_ERROR,
+      "0117",
+      "Specifying the '--coordinator' option with the '--import' option is not allowed."
+          + " Create coordinator tables separately",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:547]:
```suggestion
          + " Create Coordinator tables separately",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 7]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      "",
+      ""),
+  SCHEMA_LOADER_READING_CONFIG_FILE_FAILED(
+      Category.USER_ERROR, "0118", "Reading the config file failed. File: %s", "", ""),
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:551]:
```suggestion
      Category.USER_ERROR, "0118", "Reading the configuration file failed. File: %s", "", ""),
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 8]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+  SCHEMA_LOADER_READING_SCHEMA_FILE_FAILED(
+      Category.USER_ERROR, "0119", "Reading the schema file failed. File: %s", "", ""),
+  SCHEMA_LOADER_PARSING_SCHEMA_JSON_FAILED(
+      Category.USER_ERROR, "0120", "Parsing the schema JSON failed. Details: %s", "", ""),
+  SCHEMA_LOADER_PARSE_ERROR_TABLE_NAME_MUST_CONTAINS_NAMESPACE_AND_TABLE(
+      Category.USER_ERROR,
+      "0121",
+      "The table name must contains the namespace and the table. Table: %s",
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:559]:
```suggestion
      "The table name must contain the namespace and the table. Table: %s",
```

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 9]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      "",
+      ""),
+  SCHEMA_LOADER_PARSE_ERROR_PARTITION_KEY_MUST_BE_SPECIFIED(
+      Category.USER_ERROR, "0122", "The partition key must be specified. Table: %s", "", ""),
+  SCHEMA_LOADER_PARSE_ERROR_INVALID_CLUSTERING_KEY_FORMAT(
+      Category.USER_ERROR,
+      "0123",
+      "Invalid clustering key format. Clustering key must be in the format of 'column_name' or 'column_name ASC/DESC'."
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:567]:
```suggestion
      "Invalid clustering key format. The clustering key must be in the format of 'column_name' or 'column_name ASC/DESC'."
```

**Review by josh-wong — CHANGES_REQUESTED** [review]
Nice work!
I've added some minor suggestions, mostly for clarity. PTAL!

**Comment by brfrn169** [comment]
@josh-wong I've applied your suggestions. Could you please doublecheck just in case when you have time? Thank you.

File: core/src/main/java/com/scalar/db/common/error/CoreError.java [hunk 10]
```diff
@@ -492,6 +492,86 @@ public enum CoreError implements ScalarDbError {
...
+      "0121",
+      "The table name must contain the namespace and the table. Table: %s",
+      "",
+      ""),
+  SCHEMA_LOADER_PARSE_ERROR_PARTITION_KEY_MUST_BE_SPECIFIED(
+      Category.USER_ERROR, "0122", "The partition key must be specified. Table: %s", "", ""),
+  SCHEMA_LOADER_PARSE_ERROR_INVALID_CLUSTERING_KEY_FORMAT(
+      Category.USER_ERROR,
+      "0123",
+      "Invalid clustering key format. The clustering key must be in the format of 'column_name' or 'column_name ASC/DESC'."
```
Comment (josh-wong) [inline core/src/main/java/com/scalar/db/common/error/CoreError.java:567]:
```suggestion
      "Invalid clustering-key format. The clustering key must be in the format of 'column_name' or 'column_name ASC/DESC'."
```

**Review by josh-wong — APPROVED** [review]
I've left one very minor suggestion. Other than that, LGTM. Thank you!🙇🏻‍♂️

**Review by kota2and3kan — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**